packages/fxa-payments-server/src/store/amplitude-middleware.ts

```diff
@@ -0,0 +1,187 @@
+import { AnyAction, Dispatch, Middleware, MiddlewareAPI } from 'redux';
+import { logAmplitudeEvent } from '../lib/flow-event';
+
+const eventGroupNames = {
+  createSubscription: 'subPaySetup',
+  updatePayment: 'subPayManage',
+  cancelSubscription: 'subCancel',
+  manageSubscriptions: 'subManage',
+};
+
+const eventTypeNames = {
+  view: 'view',
+  engage: 'engage',
+  submit: 'submit',
+  success: 'success',
+  fail: 'fail',
+  complete: 'complete',
+};
+
+const getPlanPropsFromAction = (action: any) => {
+  const { plan = {} } = action.meta || action.payload || {};
+  return { planId: plan.plan_id, productId: plan.product_id };
+};
+
+const getFailureReasonFromAction = (action: any) => {
+  if (action.payload && action.payload.message) {
+    return { reason: action.payload.message };
+  }
+
+  return {};
+};
+
+export const AmplitudeMiddleware: Middleware = (
+  api: MiddlewareAPI<Dispatch<AnyAction>, any>
+) => (next: Dispatch<AnyAction>) => (action: any) => {
+  // This middleware should not break anything in dispatching actions.
+  try {
+    switch (action.type) {
+      case 'manageSubscriptionsMounted':
+        logAmplitudeEvent(
+          eventGroupNames.manageSubscriptions,
+          eventTypeNames.view,
+          {}
+        );
+        break;
+      case 'manageSubscriptionsEngaged':
+        logAmplitudeEvent(
+          eventGroupNames.manageSubscriptions,
+          eventTypeNames.engage,
+          {}
+        );
+        break;
+
+      case 'createSubscriptionMounted':
+        logAmplitudeEvent(
+          eventGroupNames.createSubscription,
+          eventTypeNames.view,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'createSubscriptionEngaged':
+        logAmplitudeEvent(
+          eventGroupNames.createSubscription,
+          eventTypeNames.engage,
+          getPlanPropsFromAction(action)
+        );
+        break;
+      case 'createSubscription_PENDING':
```

Comment:
Stick with lowerCamelCase for the new action names: 'createSubscriptionPending' or 'createSubscriptionRequestPending', not 'createSubscription_PENDING'.

Please make the same change for the others, like '_FULFILLED' and '_REJECTED'.

Reply:
Those action type are from `redux-promise-middleware`, which was why I tried to make our actions consistent with them previously.